### ISSUE TITLE
PartionMetadata Ordering

### ISSUE THREAD

**Issue by tustvold** [influxdata/influxdb_iox#1579]
A `PartitionSummary` contains a list of `TableSummary` which each contain a list of `ColumnSummary`. The ordering of columns within these is not specified, however, some codepaths expect a consistent order - most notably ChunkSnapshot assumes the order matches that of the output schema for `Selection::All`, but also a number of tests.

This feels fragile, but I'm not sure what the best solution is. It feels like we shouldn't need to rely on ordering of columns, but then it also feels off that the TableSummary would report columns in a different order to the produced arrow schema. More thought is needed

**Comment by alamb** [comment]
Yeah -- it would be nice to make some sort of invariant if `Selection:All` as specified that the columns came out in a specific order

One possible complication in SQL is that the order of column definition matters

Like if you do

```sql
create table foo (x int, y int);
create table bar (y int, x int);
```

The output of `select * from foo` has to be in `x`, `y` order, and the output of `select * from bar` has to be in `y`, `x` order

**Comment by pauldix** [comment]
Would always ordering by lexicographical ordering of the column names?

**Comment by alamb** [comment]
> Would always ordering by lexicographical ordering of the column names work?

for Line Protocol data I think it would.

If we ever support `CREATE TABLE` we may have to return the columns in the same order they were provided, I think
